### PR TITLE
feat: build an extra docker tag with commit sha

### PR DESCRIPTION
**Pull request by JorTurFer**
Signed-off-by: Jorge Turrado <jorge_turrado@hotmail.es>

This PR includes a new tag based on the git SHA for having the option of using a specific image from canary

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)

Related to https://github.com/kedacore/http-add-on/issues/567

### CODE REVIEW

**Review by abatilo — APPROVED** [review]

File: Changelog.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ This changelog keeps track of work items that have been completed and are ready
 
 ### Breaking Changes
```
Comment (tomkerkhove) [inline Changelog.md:16]: I don't think this is breaking though

File: Changelog.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ This changelog keeps track of work items that have been completed and are ready
...
 
-- TODO
+- **General**: Build an extra docker tag with commit SHA ([#567](https://github.com/kedacore/http-add-on/issues/567))
```
Comment (tomkerkhove) [inline Changelog.md:18]:
```suggestion
- **General**: Automatically tag Docker image with commit SHA ([#567](https://github.com/kedacore/http-add-on/issues/567))
```

File: Makefile [hunk 3]
```diff
@@ -7,9 +7,17 @@ IMAGE_REGISTRY 	?= ghcr.io
...
+IMAGE_INTERCEPTOR_BASE	?= ${IMAGE_REGISTRY}/${IMAGE_REPO}/http-add-on-interceptor
+IMAGE_SCALER_BASE		?= ${IMAGE_REGISTRY}/${IMAGE_REPO}/http-add-on-scaler
+
+IMAGE_OPERATOR_VERSION		?= ${IMAGE_OPERATOR_BASE}:$(VERSION)
+IMAGE_INTERCEPTOR_VERSION	?= ${IMAGE_INTERCEPTOR_BASE}:$(VERSION)
+IMAGE_SCALER_VERSION		?= ${IMAGE_SCALER_BASE}:$(VERSION)
+
+IMAGE_OPERATOR_SHA		?= ${IMAGE_OPERATOR_BASE}:$(GIT_COMMIT_SHORT)
+IMAGE_INTERCEPTOR_SHA	?= ${IMAGE_INTERCEPTOR_BASE}:$(GIT_COMMIT_SHORT)
+IMAGE_SCALER_SHA		?= ${IMAGE_SCALER_BASE}:$(GIT_COMMIT_SHORT)
```
Comment (tomkerkhove) [inline Makefile:20]: Let's correctly use "image" and "tag" concepts
```suggestion
IMAGE_OPERATOR 	?= ${IMAGE_REGISTRY}/${IMAGE_REPO}/http-add-on-operator
IMAGE_INTERCEPTOR	?= ${IMAGE_REGISTRY}/${IMAGE_REPO}/http-add-on-interceptor
IMAGE_SCALER		?= ${IMAGE_REGISTRY}/${IMAGE_REPO}/http-add-on-scaler

IMAGE_OPERATOR_VERSIONED_TAG		?= ${IMAGE_OPERATOR}:$(VERSION)
IMAGE_INTERCEPTOR_VERSIONED_TAG	?= ${IMAGE_INTERCEPTOR}:$(VERSION)
IMAGE_SCALER_VERSIONED_TAG		?= ${IMAGE_SCALER}:$(VERSION)

IMAGE_OPERATOR_SHA_TAG		?= ${IMAGE_OPERATOR}:$(GIT_COMMIT_SHORT)
IMAGE_INTERCEPTOR_SHA_TAG	?= ${IMAGE_INTERCEPTOR}:$(GIT_COMMIT_SHORT)
IMAGE_SCALER_SHA_TAG		?= ${IMAGE_SCALER}:$(GIT_COMMIT_SHORT)
```

File: Changelog.md (hunk 1 above)
Comment (JorTurFer) [reply Changelog.md:16]: it isn't indeed, not after the revert of registry changes

File: Makefile (hunk 3 above)
Comment (JorTurFer) [reply Makefile]: sure

**Review by tomkerkhove — APPROVED** [review]

**Pull request merged**